core/src/main/scala/chisel3/probe/package.scala

```diff
@@ -89,27 +102,45 @@ package object probe extends SourceInfoDoc {
   }
 
   /** Override existing driver of a writable probe on initialization. */
-  def forceInitial(probe: Data, value: Data)(implicit sourceInfo: SourceInfo): Unit = {
-    requireHasWritableProbeTypeModifier(probe, "Cannot forceInitial a non-writable Probe.")
-    pushCommand(ProbeForceInitial(sourceInfo, probe.ref, padDataToProbeWidth(value, probe).ref))
+  def forceInitial[T <: Data](probe: T, value: T)(implicit sourceInfo: SourceInfo): Unit = {
+    collectLeafMembers(probe).zip(collectLeafMembers(value)).foreach {
```

Comment:
Same comment about iterating on these together (we also need to check their types which you can do while iterating).